docs/inventories.rst

```diff
@@ -0,0 +1,773 @@
+
+For instance, you can construct groups based on these host properties:
+
+- RHEL major/minor versions
+- Windows hosts
+- Cloud based instances tagged in a certain region
+- other
+
+These examples described in the subsequent sections are organized by the structure of the input inventories.
+
+Group name and variables filtering
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Two different conditions are demonstrated here to describe the input inventory content:
+
+- First condition is that the ``state`` variable defined on the host is set to ``shutdown``
+- Second condition is membership in a group with ``account_alias`` variable set to ``product_dev``
+
+The variable ``account_alias`` is used to demonstrate a group variable. In this hypothetical, each account has its own group, with group variables giving metadata about those accounts, which is common in cloud-source inventories. These variables are shown in the general hostvars namespace in Ansible, which is why it has no special treatment in ``source_vars``.
+
+The hosts inside the input inventory will fit one condition, the other condition,
+neither, or both. This results in four hosts total for demonstration purposes.
+
+.. image:: _static/images/inventories-constructed-inventory-venn.png
+   :alt: Venn diagram describing the input inventory content for a constructed inventory
+
+
+This folder defines the inventory as an ini type named ``two_conditions.ini``:
+
+::
+
+  [account_1234]
+  host1
+  host2 state=shutdown
+
+  [account_4321]
+  host3
+  host4 state=shutdown
+
+  [account_1234:vars]
+  account_alias=product_dev
+
+  [account_4321:vars]
+  account_alias=sustaining
+
+
+The goal here is to return only shutdown hosts that are present in the group with the ``account_alias`` variable of ``product_dev``.
+There are two approaches to this, both shown in yaml format. The first one suggested is recommended.
+
+.. _constr_inv_example:
+
+1. **Construct 2 groups, limit to intersection**
+
+``source_vars``:
+
+::
+
+  plugin: constructed
+  strict: true
+  groups:
+    is_shutdown: state | default("running") == "shutdown"
+    product_dev: account_alias == "product_dev"
+
+
+``limit``: ``is_shutdown:&product_dev``
+
+This constructed inventory input creates a group for both of the categories and uses the ``limit`` (host pattern) to only return hosts that are in the intersection of those two groups, which is documented in `host patterns in Ansible <https://docs.ansible.com/ansible/latest/inventory_guide/intro_patterns.html>`_.
+
+Also, when a variable may or may not be defined (depending on the host), you can give a default, like with ``| default("running")`` if you know what value it should have when it is not defined. This helps with debugging, as described in the :ref:`constr_inv_debugging` section.
+
+2. **Construct 1 group, limit to group**
+
+``source_vars``:
+
+::
```

Comment:
syntax undeclared

Reply:
looks like this might be yaml+jinja

Reply:
I am making changes for these but haven't committed them. This was what I was doing: 
```
.. envvar:: source_vars

.. code-block::

  plugin: constructed
  strict: true
  groups:
    shutdown_in_product_dev: state | default("running") == "shutdown" and account_alias == "product_dev"


``limit``: ``shutdown_in_product_dev``

This input creates one group that only includes hosts that match both criteria. The limit is then just the group name by itself, returning just **host2**, same as the previous approach.
```

Do you suggest something else?

Reply:
What would be the best way to set that up in this context?

Reply:
Would it be something like this?
```
.. code-block:: yaml jinja
```

Reply:
I don't think there's `yaml jinja`. But I'm pretty sure `YAML+Jinja` does exist. Try it.

Reply:
File the issues for all the things you're planning to postpone for follow-ups, then. Just hiding the discussions by clicking `Resolve` doesn't actually resolve them until they are tracked on the issue tracker at least.